common/oatbox/log/StreamHandler.php

```diff
@@ -0,0 +1,96 @@
+ * if this parameter is not specified, the path to the file is taken from the system configuration
+ * for example:
+ * php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker' --log-file /var/www/html/data/tao/log/tao-nccer.log
+ *
+ * if the --log-level parameter is specified, the path to the file for logging is taken from this parameter,
+ * if this parameter is not specified, the path to the file is taken from the system configuration
+ * can take values DEBUG, INFO, NOTICE, WARNING, ERROR, CRITICAL, ALERT or EMERGENCY
+ * for example:
+ * php index.php 'oat\taoTaskQueue\scripts\tools\RunWorker' --log-level DEBUG
+ *
+ * @author Andrey Niahrou, <andrei.niahrou@taotesting.com>
+ */
+class StreamHandler extends MonologStreamHandler
+{
+    /**
+     * @param resource|string $defaultStream (Unless this parameter is specified on the command line as --log-file, otherwise it is ignored) resource where data will be output
+     * @param int $defaultLevel (Unless this parameter is specified on the command line as --log-level, otherwise it is ignored) The minimum logging level at which this handler will be triggered
+     * @param bool $bubble Whether the messages that are handled can bubble up the stack or not
+     * @param int|null $filePermission Optional file permissions (default (0644) are only for owner read/write)
+     * @param bool $useLocking Try to lock log file before doing any writes
+     *
+     * @throws \Exception                If a missing directory is not buildable
+     * @throws \InvalidArgumentException If stream is not a resource or string
+     */
+    public function __construct($defaultStream, int $defaultLevel = Logger::DEBUG, bool $bubble = true, $filePermission = null, bool $useLocking = false)
+    {
+        $stream = $this->getScriptParameter('--log-file') ?: $defaultStream;
+        $logLevel = $this->getLogLevelParameter() ?: $defaultLevel;
+        parent::__construct($stream, $logLevel, $bubble, $filePermission, $useLocking);
+    }
+
+    /**
+     * @param string $parameter
+     * @return string|null
+     */
+    private function getScriptParameter(string $parameter): ?string
+    {
+        if (!in_array($parameter, $_SERVER['argv'])) {
+            return null;
+        }
+
+        $indexValueParameter = array_search($parameter, $_SERVER['argv']) + 1;
+        return $_SERVER['argv'][$indexValueParameter];
+    }
+
+    /**
+     * @return string|null
+     * @throws \Exception
+     */
+    private function getLogLevelParameter(): ?string
+    {
+        $logLevelParameter = $this->getScriptParameter('--log-level');
+        if (!$logLevelParameter) {
+            return null;
+        }
+
+        $errorLevels = Logger::getLevels();
+        if (!isset($errorLevels[$logLevelParameter])) {
+            throw new \Exception('There is no such level of logging. Use DEBUG, INFO, NOTICE, WARNING, ERROR, CRITICAL, ALERT or EMERGENCY');
```

Comment:
Let's use the same `Logger::getLevels()` here to enumerate possible values for the parameter.